### PR TITLE
Lighthouse claim benefits documents configuration updates

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
Lighthouse benefits async job was failing due to passing parameters as query params rather than as files as specified in LH docs. https://developer.va.gov/explore/api/benefits-documents/docs?version=current

## Related issue(s)
 https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6496

## Testing done
specs. VCR added is real data, not created manually.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_document_upload_200.yml
Comment (aherzberg) [inline spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_document_upload_200.yml:1]: This is a real response generated from LH endpoint.

File: lib/lighthouse/benefits_documents/configuration.rb [hunk 1]
```diff
@@ -61,16 +61,34 @@ def post(body, document_data, lighthouse_client_id = nil, lighthouse_rsa_key_pat
...
+      }
+
+      payload[:parameters] = data
+      fn = Tempfile.new('params')
+      File.write(fn, data.to_json)
+      payload[:parameters] = Faraday::UploadIO.new(fn, 'application/json')
+
+      file = Tempfile.new(document_data[:file_name])
+      File.write(file, file_body)
+      payload[:file] = Faraday::UploadIO.new(file, 'application/pdf')
```
Comment (Tonksthebear) [inline lib/lighthouse/benefits_documents/configuration.rb:90]: I'm still shocked it requires this amount of workaround to get Faraday and this endpoint to play nicely

**Review by Tonksthebear — APPROVED** [review]
Can't believe it is (hopefully) finally working

**Review by kjduensing — APPROVED** [review]
Looks good, thanks!

File: lib/lighthouse/benefits_documents/configuration.rb (hunk 1 above)
Comment (kpethtel) [reply lib/lighthouse/benefits_documents/configuration.rb:90]: Yeah, this is definitely non-obvious. IDK why you have to write the params to a tempfile. That seems pretty unnintuitive. Maybe there's a simpler way, but hey if it works...

**Review by kpethtel — APPROVED** [review]
Nice work!

**Review by ryan-mcneil — APPROVED** [review]
LGTM

**Pull request merged**